### DISCUSSION TITLE
Screaming Snake Case Keys to Camel Case

### DISCUSSION THREAD

**Discussion by bebetoalves** [CuyZ/Valinor#425] [Q&A]
I don't know if it was a team decision and I didn't find anything about it either, but apparently the `camelCaseKeys` method doesn't work with screaming snake case keys.

```php
class User
{
    public string $firstName;
}

$source = Source::array([['FIRST_NAME' => 'John'], ['FIRST_NAME' => 'Jane']])
    ->camelCaseKeys();

$users = (new MapperBuilder())
    ->allowSuperfluousKeys()
    ->mapper()
    ->map('User[]', $source); // throws exception
```

**Comment by romm** [comment]
Better late than never: with the newly introduced converters, this should be an easy task to implement in your own app, see [common converters example in documentation](https://github.com/CuyZ/Valinor/blob/0a8c0d0574690ebf70e3959a2438dcf83ce58e5c/docs/pages/how-to/common-converters-examples.md#converting-keys-format-from-snake_case-to-camelcase).
